tests/suites/hup/tests/smoke.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 balena
+ *
+ * @license Apache-2.0
+ */
+
+'use strict';
+
+module.exports = {
+  title: 'Smoke test',
+  run: async function (test) {
+    await this.context.get().hup.initDUT(
+      this, test, this.context.get().link);
+
+    const before = await this.context.get().worker.getOSVersion(this.context.get().link);
+    test.comment(`VERSION (before): ${before}`);
+
+    await this.context.get().hup.doHUP(this, test, 'image', this.context.get().hup.payload, this.context.get().link);
+
+    test.comment(`Reducing amount of time needed for rollback-health to fail`)
+    // reduce number of failures needed to trigger rollback
+    await this.context.get().worker.executeCommandInHostOS(
+      `sed -i -e "s/COUNT=.*/COUNT=1/g" -e "s/TIMEOUT=.*/TIMEOUT=10/g" $(find /mnt/sysroot/inactive/ | grep "bin/rollback-health")`,
+      this.context.get().link,
+    );
+
+    await this.context.get().worker.rebootDut(this.context.get().link);
+
+    const after = await this.context.get().worker.getOSVersion(this.context.get().link);
+    test.comment(`VERSION (after): ${after}`);
+
+    test.comment(`Waiting for rollback-health...`);
+    await this.context.get().utils.waitUntil(async () => {
+      return (
+        (await this.context.get().worker.executeCommandInHostOS(
+          `systemctl status rollback-health.service`,
+          this.context.get().link,
+        )) !== 'active'
+      );
+    }, true);
```

Comment:
```suggestion
    }, false);
```

Reply:
Why would we ever want to break out of a wait-until loop early? We are explicitly waiting until a state is true, why exit/fail when it is false?

Reply:
we want this to be false, as you say. The `true` is there to give an exit option if there's some explicit error being thrown rather than the condition evaluating to false, that might mean that the condition could never be true.

Vipul has recently changed the default for this function to be `false` I think, so in the case of not specifying, the loop won't break unless it does the given or default number of retries.

Reply:
Yo, we chatted about this. Glad we found this before it merged.